src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -284,6 +300,7 @@ def enable_zip_deploy(cmd, resource_group_name, name, src, timeout=None, slot=No
     authorization = urllib3.util.make_headers(basic_auth='{0}:{1}'.format(user_name, password))
     headers = authorization
     headers['content-type'] = 'application/octet-stream'
+    headers['User-Agent'] = 'azure-cli:{}'.format(__version__)
```

Comment:
Can you make sure this doesn't regress/ break any behavior on webapp since this is shared code?

Reply:
I used the `az webapp` command, and it worked fine. It just adds User agent `azure-cli:<version>` to all outgoing requests. This can help us understand what requests came from azure CLI when debugging.

Reply:
For setting agent, please check out https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L499